lib/data/entity.js

```diff
@@ -380,13 +392,35 @@ const getWithConflictDetails = (defs) => {
       if ('label' in v.dataReceived && v.dataReceived.label !== defMap.get(v.version - 1).label) v.serverDiff.push('label');
 
       if (conflict) {
-        v.conflict = v.conflictingProp && v.conflictingProp.length > 0 ? ConflictType.HARD : ConflictType.SOFT;
+        v.conflict = v.conflictingProperties && v.conflictingProperties.length > 0 ? ConflictType.HARD : ConflictType.SOFT;
+
+        v.resolved = lastResolveEvent && lastResolveEvent.details.entityDefId >= def.id;
+
+        // lastGoodVersion is the baseVersion of first unresolved conflict
+        if (!v.resolved && lastGoodVersion === 0) {
+          lastGoodVersion = v.baseVersion;
 
-        v.resolve = false; // TOOD
+          // baseVersion will be always in versionMap; if not then something terrible has happened - 500 error is fine
+          versionMap.get(v.baseVersion).lastGoodVersion = true;
+        }
+      }
+
+      // We have some unresolved conflicts
+      if (lastGoodVersion > 0) {
+        relevantBaseVersions.add(v.baseVersion);
       }
     }
+
+    versionMap.set(v.version, v);
+
     result.push(v);
   }
+
+  // We return all the versions after last good version and the base versions of those.
+  if (relevantToConflict) {
+    return result.filter(v => lastGoodVersion > 0 && (v.version >= lastGoodVersion || relevantBaseVersions.has(v.version)));
```

Comment:
Could we move the `lastGoodVersion > 0` condition out of the loop?

```suggestion
  if (relevantToConflict && lastGoodVersion > 0) {
    return result.filter(v => v.version >= lastGoodVersion || relevantBaseVersions.has(v.version));
```

Reply:
Oh wait, I guess that doesn't really work, because we want to return `[]` if `lastGoodVersion === 0`. I think it'd be nice to move the static `lastGoodVersion > 0` condition out of the loop, but maybe it's not worth an extra if/ternary. 🤔

Reply:
Added inline ternary.